borsh/tests/test_generic_struct.rs

```diff
@@ -31,6 +37,14 @@ enum B<F, G> {
     Y(G),
 }
 
+/// `T: PartialOrd` bound is required for `BorshSerialize` derive to be successful
```

Comment:
Serde isn't smart enough to automatically add these bounds either. And it's not possible for user-defined types (`HashMapWithCustomBounds`). 

impl<'de, K, V, S> [Deserialize](https://docs.rs/serde/latest/serde/trait.Deserialize.html)<'de> for [HashMap](https://doc.rust-lang.org/nightly/std/collections/hash/map/struct.HashMap.html)<K, V, S>where
    K: [Deserialize](https://docs.rs/serde/latest/serde/trait.Deserialize.html)<'de> + [Eq](https://doc.rust-lang.org/nightly/core/cmp/trait.Eq.html) + [Hash](https://doc.rust-lang.org/nightly/core/hash/trait.Hash.html),
    V: [Deserialize](https://docs.rs/serde/latest/serde/trait.Deserialize.html)<'de>,
    S: [BuildHasher](https://doc.rust-lang.org/nightly/core/hash/trait.BuildHasher.html) + [Default](https://doc.rust-lang.org/nightly/core/default/trait.Default.html),
    
 ```rust
        // ERROR!
        #[derive(Debug, Deserialize)]
        struct GenericSerde<K, V>
        {
            map: HashMap<K, V>
        }
```
```bash
 1  error[E0277]: the trait bound `K: std::cmp::Eq` is not satisfied
     --> src/serialize.rs:24:18
     --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.166/src/de/mod.rs:1729:12
 2  error[E0277]: the trait bound `K: Hash` is not satisfied
     --> src/serialize.rs:24:18
     --> /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.166/src/de/mod.rs:1729:12
```
```rust
        // OK
        #[derive(Debug, Deserialize)]
        struct GenericSerde<K: Eq + Hash , V>
        {
            map: HashMap<K, V>
        }
```